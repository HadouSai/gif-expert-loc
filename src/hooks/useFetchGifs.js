import { useEffect, useState } from "react"
import { getGiffs } from "../helpers/getGifs";

/**
 * 
 * @param {string} category 
 * @returns 
 */
export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        giffs: [],
        loading: true
    })

    // cuando cambia la categorya se ejecuta
    // no pueden ser async por que experan algo sincrono
    useEffect(() => {
        getGiffs(category).then(img => setState({ loading: false, giffs: img }));
    }, [category]);

    return state;
}