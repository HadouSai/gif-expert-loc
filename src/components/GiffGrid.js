import React from "react";
import PropTypes from "prop-types";
import GiffGridItem from "./GiffGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

/**
 *
 * @param {*} category
 * @returns
 */
const GiffGrid = ({ category }) => {
  const { giffs, loading } = useFetchGifs(category);

  return (
    <div>
      <h2>{category}</h2>

      <ol>
        {loading && <p>Loading</p>}

        {giffs.map((giff) => (
          <GiffGridItem
            key={giff.id}
            id={giff.id}
            title={giff.title}
            imgUrl={giff.url}
          />
        ))}
      </ol>
    </div>
  );
};

GiffGrid.propTypes = {
  category: PropTypes.string,
};

export default GiffGrid;
