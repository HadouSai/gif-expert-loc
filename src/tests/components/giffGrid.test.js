import "@testing-library/jest-dom";
import GiffGrid from "../../components/GiffGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe('Testing in giffgrid', () => {
    const gifs = [{
        id: 'abc',
        url: 'http://www.vaaa/ss/fs.jpg',
        title: 'pinto'
    }, {
        id: 'abc',
        url: 'http://www.vaaa/ss/fs.jpg',
        title: 'pinto'
    }]

    test('should appear GiffGrid', () => {
        useFetchGifs.mockReturnValue({
            giffs: [],
            loading: true
        });
        // voy aca el mock se me cargo todo .....
        const category = 'kurumi';
        const wrapper = shallow(<GiffGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when loaded', () => {

        useFetchGifs.mockReturnValue({
            giffs: gifs,
            loading: false
        })
        const category = 'kurumi';
        const wrapper = shallow(<GiffGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GiffGridItem').length).toBe(gifs.length)


        //expect(wrapper).toMatchSnapshot();
    })


});
