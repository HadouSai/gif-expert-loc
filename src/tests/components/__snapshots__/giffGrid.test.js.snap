// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in giffgrid should appear GiffGrid 1`] = `
<div>
  <h2>
    kurumi
  </h2>
  <ol>
    <p>
      Loading
    </p>
  </ol>
</div>
`;
